src/components/ha-file-upload.ts

```diff
@@ -140,93 +188,121 @@ export class HaFileUpload extends LitElement {
-          inset-inline-end: initial !important;
-          direction: var(--direction);
-        }
-        .mdc-text-field__icon--trailing {
-          pointer-events: auto !important;
-        }
-        .dragged:before {
-          position: var(--layout-fit_-_position);
-          top: var(--layout-fit_-_top);
-          right: var(--layout-fit_-_right);
-          bottom: var(--layout-fit_-_bottom);
-          left: var(--layout-fit_-_left);
-          background: currentColor;
-          content: "";
-          opacity: var(--dark-divider-opacity);
-          pointer-events: none;
-          border-radius: 4px;
-        }
-        .value {
-          width: 100%;
-        }
-        input.file {
-          display: none;
-        }
-        img {
-          max-width: 100%;
-          max-height: 125px;
-        }
-        ha-icon-button {
-          --mdc-icon-button-size: 24px;
-          --mdc-icon-size: 20px;
-        }
-        ha-circular-progress {
-          display: block;
-          text-align-last: center;
-        }
-      `,
-    ];
+    return css`
+      :host {
+        display: block;
+        height: 240px;
+      }
+      :host([disabled]) {
+        pointer-events: none;
+        color: var(--disabled-text-color);
+      }
+      .container {
+        position: relative;
+        display: flex;
+        flex-direction: column;
+        justify-content: center;
+        align-items: center;
+        border: solid 1px
+          var(--mdc-text-field-idle-line-color, rgba(0, 0, 0, 0.42));
+        border-radius: var(--mdc-shape-small, 4px);
+        height: 100%;
+      }
+      label.container {
+        border: dashed 1px
+          var(--mdc-text-field-idle-line-color, rgba(0, 0, 0, 0.42));
+        cursor: pointer;
+      }
+      :host([disabled]) .container {
+        border-color: var(--disabled-color);
+      }
+      label.dragged {
+        border-color: var(--primary-color);
+      }
+      .dragged:before {
+        position: absolute;
+        top: 0;
+        right: 0;
+        bottom: 0;
+        left: 0;
+        background-color: var(--primary-color);
+        content: "";
+        opacity: var(--dark-divider-opacity);
+        pointer-events: none;
+        border-radius: var(--mdc-shape-small, 4px);
+      }
+      label.value {
+        cursor: default;
+      }
+      label.value.multiple {
+        justify-content: unset;
+        overflow: auto;
+      }
+      .highlight {
+        color: var(--primary-color);
+      }
+      .row {
+        display: flex;
+        width: 100%;
+        align-items: center;
+        justify-content: space-between;
+        padding: 0 16px;
+        box-sizing: border-box;
+      }
+      ha-button {
+        margin-bottom: 4px;
+      }
+      .supports {
+        color: var(--secondary-text-color);
+        font-size: 12px;
+      }
+      :host([disabled]) .secondary {
+        color: var(--disabled-text-color);
+      }
+      input.file {
+        display: none;
```

Comment:
This is a problem - it basically nullifies all the native control you get from the `input`.  This is why the `for` attribute is needed, and also why  manual `click` events are needed to open the file browser.  There's a better way to do this.